### PR TITLE
Change the registry to use world UUIDs

### PR DESCRIPTION
**Pull request by gecko10000**
## Description
I have recently run into an issue where a world, when created quickly enough after deletion of a world with the same name, will not store data. I have tested this extensively and narrowed the issue down to [this](https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java#L102) ticker code and [this](https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/me/mrCookieSlime/Slimefun/api/BlockStorage.java#L116) check for the world's existence. Since the ticker code only runs every half second, it does not remove the world from the registry until up to a Slimefun tick later and therefore causes issues in the constructor.

An easy way to recreate this issue is to delete a world and immediately create a new one with the same name.

## Proposed changes
This PR changes the registry to use UUIDs instead of the world names themselves. This _does not_ affect the storage location in files and therefore these are not breaking changes.

This may not be the ideal way to do it; it may be better to remove the world from the registry [here](https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/WorldListener.java#L28) instead.

## Related Issues (if applicable)
No issue opened

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.14.* - 1.19.*).
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java [hunk 1]
```diff
@@ -1,12 +1,7 @@
 package io.github.thebusybiscuit.slimefun4.implementation;
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:4]: No star imports please (very possible your IDE did this automatically)

**Review by svr333 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java (hunk 1 above)
Comment (gecko10000) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:4]: Yes, it did.

**Review by svr333 — APPROVED** [review]
This does look indeed like it isnt a breaking change and would still work with all worlds.
So in that case it looks good to me!

**Comment by J3fftw1** [comment]
im not sure how i closed this throught git.
I was updating his branch.

**Pull request closed without merging**